src/python/pants/backend/python/target_types.py

```diff
@@ -720,11 +734,16 @@ class PythonTestsGeneratorTarget(TargetFilesGenerator):
     core_fields = (
         PythonTestsDependenciesField,
         PythonTestsGeneratingSourcesField,
+        InterpreterConstraintsField,
         PythonTestsOverrideField,
     )
     generated_target_cls = PythonTestTarget
-    copied_fields = (PythonTestsDependenciesField,)
+    copied_fields = (
+        PythonTestsDependenciesField,
+        InterpreterConstraintsField,
```

Comment:
We will fail (with a useful error message) to `parametrize` any core field of a target generator, and so in order to parametrize `interpreter_constraints`, we'd need to make `InterpreterConstraintsField` a `moved` field here. But there were tests (and maybe implementations?) relying on the current behavior of "all" targets in the graph having the field, so it felt best to tackle it in a followup.